traffic_server/plugins/astats_over_http/astats_over_http.c

```diff
@@ -570,22 +574,28 @@ static bool is_ip_allowed(const config_t* config, const struct sockaddr* addr) {
 	if(!addr) {
 		return true;
 	}
-	const char *ip = addr->sa_data+2;
 
 	if (addr->sa_family == AF_INET && config->allowIps) {
+		const struct sockaddr_in* addr_in = (struct sockaddr_in*) addr;
+		const char *ip = (char*) &addr_in->sin_addr;
+
 		for(i=0; i < config->ipCount; i++) {
-			ipmask = config->allowIps+ (i*5);
+			ipmask = config->allowIps + (i*5);
```

Comment:
@serDrem you should fix the ipv4 stuff too.  Besides just being bad, (I * 5) is meaningless to the reader.  Should always use sizeof() as thats the correct way to code this plus "i * sizeof(struct in_addr) + 1" lets the reader understand whats being calculated.

Reply:
done